packages/core/src/state-management/repository.ts

```diff
@@ -824,7 +824,12 @@ export class Repository {
    * Adds the stream's RunningState to the in-memory cache and subscribes the Repository's global feed$ to receive changes emitted by that RunningState
    */
   private _registerRunningState(state$: RunningState): void {
-    state$.subscribe(this.feed.aggregation.documents)
+    state$
+      .pipe(
+        distinct((s) => s.log[s.log.length - 1].cid.toString()), // Distinct if the tip changes
```

Comment:
Hmm, looking at the documentation for the `distinct` function (https://rxjs.dev/api/index/function/distinct), it looks like it uses a Set internally to keep track of *all* previous values emitted.  We actually only really need to keep track of the single most recently emitted value.  If the stream state got reset to some older value in the log (which can happen if the CACAOs expire on newer log entries and the stream is reloaded with the ALWAYS_SYNC flag), we probably would want to emit an event for that.  I'm also worried that maintaining this hash set for all streams could be a meaningful amount of additional memory consumption.  Can we do this in a way that there's a constant amount of memory consumption needed, since it only needs to remember the previous tip, not *all* previous tips?

Reply:
True. Changed to `distinctUntilKeyChanged` which does not suffer like this.